documents/Libraries/stdlib/sx-glsl/lib/sx_sampling.glsl

```diff
@@ -29,70 +32,102 @@ vec3 sx_normal_from_samples_sobel(float S[9], float _scale)
    return (norm + 1.0) * 0.5;
 };
 
-//
-// Blur using box filter for float samples
-//
-float sx_blur_box_float(float S[9])
-{
-    return (S[0] + S[1] + S[2] + S[3] + S[4] + S[5] + S[6] + S[7] + S[8]) / 9.0;
-}
-
-//
-// Blur using box filter for vec2 samples
-//
-vec2 sx_blur_box_vec2(vec2 S[9])
+// Kernal weights for box filter
+void sx_get_box_weights(inout float W[SX_MAX_SAMPLE_COUNT], int filterSize)
 {
-    return (S[0] + S[1] + S[2] + S[3] + S[4] + S[5] + S[6] + S[7] + S[8]) / 9.0;
+    int filterSize2 = filterSize*filterSize;
```

Comment:
Maybe call it filterSizeSquared?

Reply:
Or you seem to call it sampleCount elsewhere.

Reply:
Sure I can call this sampleCount